### ISSUE TITLE
method not allowed while trying to get `/metrics`

### ISSUE THREAD

**Issue by hhromic** [Kong/kong-plugin-prometheus#74]
Hi, I have a fresh installation of Kong (backed with Postgres) and after enabling the Prometheus plugin globally I am unable to retrieve any metrics using the Kong Admin API.

I'm getting a "method not allowed" (code 405) error from Kong:
```
$ http GET http://kong:8001/metrics
HTTP/1.1 405 Not Allowed
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 32
Content-Type: application/json; charset=utf-8
Date: Tue, 10 Dec 2019 17:42:49 GMT
Server: kong/1.4.1
X-Kong-Admin-Latency: 0

{
    "message": "Method not allowed"
}
```

I am also seeing the following error in Kong's logs:
```
2019/12/10 17:42:49 [error] 31#0: *21344 lua coroutine: runtime error: don't know how to respond to GET
stack traceback:
coroutine 0:
        [C]: ?
coroutine 1:
        [C]: in function 'resume'
        /usr/local/share/lua/5.1/lapis/application.lua:397: in function 'handler'
        /usr/local/share/lua/5.1/lapis/application.lua:130: in function 'resolve'
        /usr/local/share/lua/5.1/lapis/application.lua:167: in function </usr/local/share/lua/5.1/lapis/application.lua:165>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/lapis/application.lua:173: in function 'dispatch'
        /usr/local/share/lua/5.1/lapis/nginx.lua:230: in function 'serve'
        /usr/local/share/lua/5.1/kong/init.lua:1122: in function 'admin_content'
        content_by_lua(nginx-kong.conf:195):2: in main chunk, client: 10.0.41.4, server: kong_admin, request: "GET /metrics HTTP/1.1", host: "kong:8001"
10.0.41.4 - - [10/Dec/2019:17:42:49 +0000] "GET /metrics HTTP/1.1" 405 32 "-" "HTTPie/1.0.3"
```

I am running Kong in a Docker container from the official Kong image and I defined a simple service with a single route to it for testing. What can be wrong? Thanks in advance for any help.

**Comment by hbagdi** [comment]
You are running into https://github.com/Kong/kong/issues/5291.
Kong 1.4.2 is being released as I write this comment. Expect a Docker image to be up soon.

**Comment by hhromic** [comment]
Oh! It didn't occur to me to search on the Kong issue tracker itself as other plugins seem to be working for me. Glad to know the problem was identified and fixed already!
Will close this issue as soon as I confirm the new version being okay. Thanks!

**Comment by hhromic** [comment]
I updated Kong to version `1.4.2` today and I can confirm now that the prometheus plugin is working under the Admin API. In the meantime I also discovered the Status API which I believe is better suited for this plugin so I've been using that API instead of exposing `/metrics` from the Admin API.

Anyway, closing this as resolved! Thanks!

**Comment by ChenZhaobin** [comment]
@hhromic  Hi buddy ,did you mean  it can be solved  by  upgrading to 1.4.2  and change status_listen to  0.0.0.0:8001?
